mars/learn/contrib/lightgbm/_train.py

```diff
@@ -230,23 +232,25 @@ def tile(cls, op: "LGBMTrain"):
         data = op.data
         worker_to_args = defaultdict(dict)
 
-        workers = cls._get_data_chunks_workers(ctx, data)
+        # Note: Mars worker is band address, and LGBMTrain worker is machine ip.
+        ip_to_worker = cls._get_data_chunks_workers(ctx, data)
+        ips = ip_to_worker.keys()
 
         for arg in ["_data", "_label", "_sample_weight", "_init_score"]:
             if getattr(op, arg) is not None:
                 for worker, chunk in cls._concat_chunks_by_worker(
-                    getattr(op, arg).chunks, workers
+                    getattr(op, arg).chunks, ips
                 ).items():
                     worker_to_args[worker][arg] = chunk
 
         if op.eval_datas:
             eval_workers_list = [
-                cls._get_data_chunks_workers(ctx, d) for d in op.eval_datas
+                cls._get_data_chunks_workers(ctx, d).keys() for d in op.eval_datas
             ]
             extra_workers = reduce(
                 operator.or_, (set(w) for w in eval_workers_list)
-            ) - set(workers)
-            worker_remap = dict(zip(extra_workers, itertools.cycle(workers)))
+            ) - set(ips)
```

Comment:
Is it possible that multiple workers share one same ip?

Reply:
Doesn't matter. It will remove duplicate ip.